### DISCUSSION TITLE
Images Not Indexing

### DISCUSSION THREAD

**Discussion by CousinDanny** [photoprism/photoprism#3886] [Q&A] [answered]
Hi! Sorry in advance, this is very new to me. I'm running Photoprism via Docker Compose on Windows 10 Pro. I have created a container and added a local volume but when I run a full library scan only 65 out of 100s of my photos and videos are recognized. I don't see any error logs. I've looked through a dozen posts about people having similar issues but nothing seems to help. I've tried deleting and restarting the container. I've tried adding and removing test volumes in the YML file to try and trick it into recognizing new images. I've changed my permissions in the Windows Registry. WSL has 16 gigs of RAM available. I'm really at a loss but would love to get this working properly. Any advice would be greatly appreciated. Thanks!

**Comment by graciousgrey** [comment]
Hey :)

There may be several reasons for this, e.g. the photos may be duplicates, stacked or under review. You can find more reasons in our troubleshooting guide: https://docs.photoprism.app/getting-started/troubleshooting/#missing-pictures

**Reply by CousinDanny** [reply]
Hi thank you for the reply! I went through the troubleshooting doc and annotated it based on what I see, with screenshots attached. 

> Missing Pictures
> 
> If you have indexed your library and some images or videos are missing, first check Library > Errors for errors and warnings. In case the application logs don't contain anything helpful:
> **No errors**
> 
>     The files exceed the size limit in megabyte or the resolution limit in megapixels
> **File size limits disabled in YML file**
>     The files have bad filesystem permissions or the wrong owner, so they cannot be opened
> **Not sure how to check but seems unlikely, they’re mostly JPEGs from my phone**
>     The pictures are in Review due to low quality or incomplete metadata
> **There were some files in review, so I disabled quality filter, although there are still many more missing**
>     The file type is generally unsupported
> **Mostly JPEGs, PNG and MP4**
>     The file type is generally supported, but a specific feature or codec is not
> **Mostly JPEGs, PNG and MP4**
>     The indexer has skipped the files because they are exact duplicates
> **Not duplicates**
>     The files are ignored based on pattern in a .ppignore file
> **Have not created .ppignore file**
>     They are in Library > Hidden because thumbnails could not be created:
> **No files in Hidden**
>         Convert to JPEG is disabled in Settings > Library (enable it)
> **Don’t see anything about converting to JPEG in Settings>Library?**
>         FFmpeg and/or RAW converters are disabled in Settings > Advanced
> **Not disabled in Settings>Advanced**
>         The file is broken, e.g. because of short Huffman data (try to fix it)
> **Not sure but seems unlikely**
>         Your (virtual) server disk is full (add storage)
> **Plenty of storage**
>         The storage folder is not writable (change permissions)
> **Should be writable. I didn’t specify read-only intentionally. Not sure how to check?**
>         A disk usage or the inode limit has been reached (remove or increase it)
> **Seems unlikely...**
>     Multiple files were stacked based on their metadata or file names
> **Unique filenames all the way down**
>     The private or archived status was restored from a backup
> **Not the case**
>     The NSFW filter is enabled, so they were marked as private
> **NSFW filter disabled, no images in private**
>     You are not signed in as admin, so you can't see everything
> **Signed in as admin**
>     You try to index a shared drive on a remote server, but the server is offline
> **All local drives**
>     The indexer has crashed because you didn't configure at least 4 GB of swap
> **More than 4GB of RAM**
>     Somebody has deleted files without telling you
> **All photos and video are accessible and viewable in their original folder location via Windows Explorer**
>     You are connected to the wrong server, VPN, CDN, or a DNS record has not been updated yet
> **I don’t understand this one exactly but I have only done the basic steps to create a PP container so seems unlikely**
> 
> Depending on the cause of the problem, you may need to perform a full rescan once the issue is resolved.
> **

It could definitely be a more fundamental thing like I set up the server wrong or maybe a Docker mistake. I just don't know how else to investigate at this point. :-(

![firefox_ivI2q0cYr3](https://github.com/photoprism/photoprism/assets/150866686/b0456924-187b-49f6-b0a4-4f6c773e13c6)
![firefox_tjIS20znCE](https://github.com/photoprism/photoprism/assets/150866686/300bf32a-dd35-4c94-ada6-e9afa34050d2)
![notepad_z0R5jaCQ3s](https://github.com/photoprism/photoprism/assets/150866686/bac5719e-aea1-4ac2-89b0-e27e8b42344f)
![soffice bin_ddGuAKrV5W](https://github.com/photoprism/photoprism/assets/150866686/5bff5f31-55cd-4d4c-93ac-16ae046b85a9)
![ubuntu_c2lxFc0K2Z](https://github.com/photoprism/photoprism/assets/150866686/6aee27e7-4521-4cc5-b875-ca71ed79af0d)
![explorer_2TnOfAVPt0](https://github.com/photoprism/photoprism/assets/150866686/6d78686b-5626-4c94-a43d-fd039186b96c)**

**Reply by CousinDanny** [reply]
Quick follow up: In the attached images you'll see all folders that should be recognized and added vs. the four that actually get indexed. And even within the four recognized folders, only a few images are indexed. I tried creating a test directory with only a few random photos in there to see if they would get picked up but that didn't work either.
![notepad_J1AjJHpDg9](https://github.com/photoprism/photoprism/assets/150866686/fc877de3-b345-4f15-99ca-5956289cc902)
![firefox_AcwdL9OIGA](https://github.com/photoprism/photoprism/assets/150866686/26126bf2-a752-4c5f-8af1-717607b14b40)
![explorer_8GdUEVFmcO](https://github.com/photoprism/photoprism/assets/150866686/4ed63645-5fde-4cc8-9599-08da0f496399)

**Reply by lastzero** [reply]
Please closely folllow our documentation for mounting multiple folders:
- https://docs.photoprism.app/getting-started/troubleshooting/docker/#overlay-volumes

**Reply by CousinDanny** [reply]
Again I'm new to this, but I tried to follow the documentation which if I understand it correctly, says to execute the code it links to. It seems to have pulled and run successfully. I restarted PP and reindexed but it's the same behavior. I'm probably missing some key concept but sorry I just don't know what it is.

![ubuntu_5rJFsz7WN3](https://github.com/photoprism/photoprism/assets/150866686/7e2e26f9-d7c3-41c4-8708-6f52c70818ae)

**Reply by lastzero** [reply, marked as answer]
The problem is that you are mounting ALL directories under `/photoprism/originals` instead of using DIFFERENT directories as shown in our documentation, e.g. `photoprism/originals` and `/photoprism/originals/media`. For this reason, not all images can be indexed.

**Reply by CousinDanny** [reply]
Wow ok I guess that's the issue. I was under the impression that I was supposed to take all my directories and point them to photoprism/originals. Just so I understand, any time I want to add a new media library I should add them to photoprism/originals/#whatever-sub-folder?
